src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -225,8 +227,41 @@ public CodeGenerator getCodeGenerator(Format outputFormat, CodeConsumer cc) {
       }
     }
 
+    result.moduleRewriteLog = generateModuleRewriteLog(filesToConvert, modulePrePass);
     ((GentsErrorManager) compiler.getErrorManager()).doGenerateReport();
-    return sourceFileMap;
+    return result;
+  }
+
+  /**
+   * Generates a log file that maps how goog.modules/goog.provides where mapped to TS modules.
+   *
+   * <p>Each line of the log contains three comma separated strings - (OriginalName, TSFile,
+   * DefaultExportName).
+   *
+   * <p>OriginalName is the original goog.module/goog.provide name. TSFile - the name of the
+   * TypeScript file that contains the converted Closure module/namespace. DefaultExportName - Gents
+   * never uses TS default exports. This means if the original goog.module had a default export we
+   * generated a named export with a specific name. This field contains that name. If there was no
+   * default export, the field contains the empty string.
+   *
+   * <p>Ex: (in file buz.js) goog.module(foo.bar) class A { ... } exports = A;
+   *
+   * <p>is translated to buz.ts: export class A { ... }
+   *
+   * <p>Which generates the following log line: foo.bar,buz.ts,A
+   */
+  private String generateModuleRewriteLog(
+      Set<String> filesToConvert, CollectModuleMetadata modulePrePass) {
+    String output = "";
+    for (String s : modulePrePass.getNamespaceMap().keySet()) {
+      FileModule tsModule = modulePrePass.getNamespaceMap().get(s);
+      String file = tsModule.file;
+      String defaultRename = tsModule.exportedNamespacesToSymbols.getOrDefault("exports", "");
+      if (filesToConvert.contains(file)) {
+        output += (s + "," + file + "," + defaultRename + "\n");
```

Comment:
Appending to a `String` in Java creates a new object, so this is O(n^2). Consider using a `StringBuilder`.

Here's much more than you need to know about it:
http://www.pellegrino.link/2015/08/22/string-concatenation-with-java-8.html